rfd/0208-selinux-policy.md

```diff
@@ -0,0 +1,188 @@
+If `teleport-update` is unable to install the module, it will rollback to the module that was previously installed, if any.
+
+`teleport-update` will only manage the official Teleport SELinux module. Modules created by users or third parties will not be inspected or
+modified. If an SELinux module installed that shares a name with the official Teleport SELinux module it will be assumed to be a version of
+the Teleport SELinux module and may be modified.
+
+If the SSH service itself was allowed to manage its own SELinux module, an attacker that compromised the SSH service could modify the module to
+be more permissive, defeating the point of the module in the first place.
+
+Installation scripts will be updated to install and configure the SELinux module with the `teleport-update`.
+
+SELinux has 3 modes, disabled, permissive (module violations are only logged) and enforcing (attempts to violate the module are denied).
+The SELinux mode will not be modified by `teleport-update`, users will be responsible for enabling SELinux if they so choose.
+
+The `teleport-update` will have a `--dry-run` flag added to the necessary subcommands that will print what actions it would need to take to
+manage the module if run normally and print the module that would be installed based on the provided Teleport agent configuration file.
+
+The following RPM packages will be required to install the SELinux module:
+- selinux-policy-devel
+- policycoreutils
+
+#### Examples
+
+```
+# installing a module
+$ teleport-update enable --enable-selinux
+SELinux module installed successfully
+# with --dry-run
+$ teleport-update enable --enable-selinux --dry-run
+<SELinux module contents>
+> checkmodule -m teleport.te -o teleport.mod
+> semodule_package -o teleport.pp -m teleport.mod
+> semodule -i teleport.pp
+```
+
+### Compliance
+
+To ensure the SSH service SELinux module is installed and SELinux is configured to enforce it, a `--selinux` flag will be added to the `teleport start` subcommand
+of Teleport agents. If the SELinux module is not installed or SELinux is not present and in enforcing mode and `--selinux` is passed, the Teleport agent
+will exit with an error.
+
+Additionally since the SSH service is the only Teleport agent service that will be initially supported by the module, if the `--selinux` flag is passed
+and the Teleport agent does not have the SSH service enabled, or at least one other Teleport service is enabled then the Teleport agent will exit with an error.
+Additionally if there is a SELinux module for Teleport agents installed but it does not match the module that would be generated with the version of the
+module embedded in the Teleport agent and the Teleport agent's configuration file it will notify the user and exit with an error. Users will then have to
+run `teleport-update` to update the SELinux module.
+
+### UX
+
+A `--troubleshoot` flag will be added to the `teleport selinux` subcommand that will attempt to use the `audit2why` command
```

Comment:
What will the command do? Just wondering if it should be a flag or a subcommand.